lib/gollum/public/gollum/stylesheets/template.scss.erb

```diff
@@ -461,6 +466,32 @@ a {
       padding: 0;
     }
   }
+
+  .btn-copy {
```

Comment:
I don't really know how to use `primer.css`, if you could point me in some direction maybe we can reduce the amount of CSS.

Reply:
This seems to do the trick. In `gollum.js.erb`:
```javascript
button.classList.add("btn", "btn-large", "btn-copy", "position-absolute", "top-2", "right-2");
```

Then in `template.scss.erb` you only need:
```css
  .btn-copy {
    background-image:  url('data:image/svg+xml;utf8,<%= rocticon_css(:copy) %>');
    background-repeat: no-repeat;
    background-position: center;
    transition: all 200ms ease-in;
  }

  .btn-copy.active {
    background-image:  url('data:image/svg+xml;utf8,<%= rocticon_css(:check) %>');
   }
```

Reply:
This replicates the current behavior, I think.
![Screenshot 2022-11-16 at 09 37 57](https://user-images.githubusercontent.com/571173/202130231-fbc45c18-9529-4348-9f5a-d8ab8885e7fb.png)

However, looking at how Github does it, it should be possible to have the width of the button "shrink" around the icon, like so:
![Screenshot 2022-11-16 at 09 37 33](https://user-images.githubusercontent.com/571173/202130140-7534d189-a4f9-482a-b174-82612e0a1ea7.png)

However, they seem to have the `svg` as innerHTML for the button, and not as a background image, so that might explain the difference.

Reply:
Two other points to think about:
* On Github, the copy to clipboard icon is also shown in the Preview pane. I think we also want to replicate this, right @dometto @benjaminwil ?
* On Github, the copy to clipboard icon is _only_ shown on mouseover. Do we want to replicate that as well?

Reply:
I think we can even do without any CSS if we'd just change the `innerHTML` of the button on click. I haven't tried this yet, but something like `target.innerHTML = '<%= rocticon(:check) %>';` should work.

Reply:
I updated the code and removed all custom CSS except for the `position: relative` to `pre` elements, works fine but the icon is not that centered??

![image](https://user-images.githubusercontent.com/57039432/202602534-1bb95f98-060e-4c0d-98d0-074ca9fc1228.png)

I give it a try to add the copy button to the Preview pane but I couldn't make it work, I look into it more later. Let me know if you have any hints or if this is not needed in the end.

Reply:
I managed to change the button to the size I would expect:
![Screenshot 2022-11-18 at 09 54 19](https://user-images.githubusercontent.com/571173/202662336-70298f48-78a6-4c38-8068-a5ab4013cb55.png)

Reply:
We can further add a tooltip, just like Github does: 

![Screenshot 2022-11-18 at 10 11 17](https://user-images.githubusercontent.com/571173/202667304-1ae5ddc5-ee4c-4cd4-9508-0a8eeeccf4d7.png)

```javascript
target.innerHTML = iconCopyActive;
        target.classList.add("tooltipped", "tooltipped-no-delay", "tooltipped-w");
        target.setAttribute('aria-label', 'Copied!');
        // wait 1 secconds
        await new Promise(resolve => setTimeout(resolve, 1000));
        target.innerHTML = iconCopyDefault;
        target.classList.remove("tooltipped", "tooltipped-no-delay", "tooltipped-w");
        target.setAttribute('aria-label', 'copy to clipboard');
```

Reply:
Hey! it's looks awesome!

I implemented it right away, I just increase it the wait time to 2 seconds so you can read the tooltip.

Reply:
Thanks! After a hint from @dometto I briefly looked into getting the buttons working in the preview tab. The "problem" is that preview content (including the code blocks) is added to the DOM dynamically, which means that the Javascript is not yet attached for that content. The solution should be to wrap the javascript in a function that we can call after the preview content is loaded. This bit is already set up [here](https://github.com/gollum/gollum/blob/master/lib/gollum/public/gollum/javascript/gollum.js.erb#L375), so it should be straightforward to add a call to a new function (e.g., `enable_copy_code_buttons`) there. Does that make sense @aleee-idk ?

Reply:
Thanks! I was searching where I needed to call the function but couldn't find it, it's works well.